src/robusta/core/discovery/discovery.py

```diff
@@ -478,3 +592,22 @@ def extract_volumes(resource) -> List[V1Volume]:
     except Exception:  # may fail if one of the attributes is None
         logging.error(f"Failed to extract volumes from {resource}", exc_info=True)
     return []
+
+
+def extract_volumes_k8(resource) -> List[Volume]:
+    """Extract volumes from k8s python api object (not hikaru)"""
+    try:
+        volumes = []
+        if (
```

Comment:
to avoid code duplication we may need to use the spec as we discussed earlier. But for V1Pod, V1Deployment etc the spec is `V1PodSpec` while for the Pod, Deployment etc the spec is `PodSpec`.

Either we continue with the duplicated two lines of code or can ignore the return types of the spec and handle it within the exract_volumes and extract_volumes_k8 methods

Reply:
As per our discussion, it is difficult to de-duplicate the code due to inconsistent return types.